src/concurrency/vector_clock.rs

```diff
@@ -313,7 +368,14 @@ impl Index<VectorIdx> for VClock {
 
     #[inline]
     fn index(&self, index: VectorIdx) -> &VTimestamp {
-        self.as_slice().get(index.to_u32() as usize).unwrap_or(&0)
+        self.as_slice().get(index.to_u32() as usize).unwrap_or(&VTimestamp::NONE)
+    }
+}
+
+impl IndexMut<VectorIdx> for VClock {
```

Comment:
Side-note: we could probably use https://doc.rust-lang.org/nightly/nightly-rustc/rustc_index/vec/struct.IndexVec.html for `VClock` and https://doc.rust-lang.org/nightly/nightly-rustc/rustc_index/macro.newtype_index.html for `VectorIdx`

Reply:
Hm, `IndexVec` just wraps a `Vec` and currently this uses `SmallVec`, and this PR is already messing with the inline capacity. There are ambient concerns about the data race detector overhead: https://github.com/rust-lang/miri/issues/2716 https://github.com/rust-lang/miri/issues/1689 which in my experience are not backed up by profiling, but I still think the prudent thing to do is adjust one thing at a time here. Happy to look into this in a separate PR.

Reply:
Makes sense